frontend/src/components/SamplesList.tsx

```diff
@@ -141,25 +148,10 @@ export default function SamplesList({
       );
       setChanges(updatedChanges);
       if (updatedChanges.length === 0) setUnsavedChanges?.(false);
+      resetAlertIfCostCentersAreAllValid(updatedChanges);
       return;
     }
 
-    // validate Cost Center inputs
-    if (fieldName === "costCenter") {
-      if (!isValidCostCenter(newValue)) {
-        setAlertContent(costCenterAlertContent);
-        setShowAlertModal(true);
-      } else {
-        const allRowNodesInView = rowNode.parent?.allLeafChildren;
-        const allRowsHaveValidCostCenter = allRowNodesInView?.every(
-          (rowNode) =>
-            rowNode.data?.costCenter &&
-            isValidCostCenter(rowNode.data.costCenter)
-        );
-        if (allRowsHaveValidCostCenter) setAlertContent(defaultAlertContent);
-      }
-    }
```

Comment:
Moved this block of code down below, to be after where we update the `changes` state array, because it relies on this `changes` state array to check whether all the edited Cost Center fields have a valid format.